sync_check/config.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/BurntSushi/toml"
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb-tools/sync_check/util"
+)
+
+// NewConfig creates a new config.
+func NewConfig() *Config {
+	cfg := &Config{}
+	cfg.FlagSet = flag.NewFlagSet("sync-check", flag.ContinueOnError)
+	fs := cfg.FlagSet
+
+	fs.StringVar(&cfg.ConfigFile, "config", "", "Config file")
+	fs.StringVar(&cfg.LogLevel, "L", "info", "log level: debug, info, warn, error, fatal")
+	fs.IntVar(&cfg.Delay, "delay", 0, "check data 5 second before")
+	fs.IntVar(&cfg.ChunkSize, "chunk-size", 1000, "diff check chunk size")
+	fs.IntVar(&cfg.Sample, "sample", 100, "the percent of sampling check")
+	fs.IntVar(&cfg.CheckThCount, "check-thcount", 1, "the count of check thread count")
+	fs.StringVar(&cfg.TimeField, "time-field", "", "a field with datetime type")
+	fs.StringVar(&cfg.BeginTime, "begin-time", "", "check data's begin time")
+	fs.StringVar(&cfg.EndTime, "end-time", "", "check data's end time")
+	fs.StringVar(&cfg.SplitField, "split-field", "", "use this field split data to several chunk")
+
+	return cfg
+}
+
+// Config is the configuration.
+type Config struct {
+	*flag.FlagSet `json:"-"`
+
+	LogLevel string `toml:"log-level" json:"log-level"`
+
+	TimeField string `toml:"time-field" json:"time-field"`
+
+	BeginTime string `toml:"begin-time" json:"begin-time"`
+
+	EndTime string `toml:"end-time" json:"end-time"`
+
+	SplitField string `toml:"split-field" json:"split-field"`
+
+	SourceDBCfg util.DBConfig `toml:"source-db" json:"source-db"`
+
+	TargetDBCfg util.DBConfig `toml:"target-db" json:"target-db"`
+
+	ConfigFile string
+
+	Delay int `toml:"delay" json:"delay"`
+
+	ChunkSize int `toml:"chunk-size" json:"chunk-size"`
+
+	Sample int `toml:"sample" json:"sample"`
+
+	CheckThCount int `toml:"check-thcount" json:"check-thcount"`
```

Comment:
what's `Th` for ? Try to name exportable variable in detail

Reply:
th means thread